### ISSUE TITLE
Folia block 

### ISSUE THREAD

**Issue by Q2297045667** [Snowiiii/Pumpkin#320]
### I've searched existing issues and couldn't find a duplicate.

- [X] I confirm this is not a duplicate.

### Is your feature request related to a problem? Please describe.

Is there any plan in the future to consider incorporating Folia's block system? In my opinion, Rust's ownership model is ideally suited for this task.

## Overview

Folia groups nearby loaded chunks to form an "independent region."
See [the PaperMC documentation](https://docs.papermc.io/folia/reference/region-logic) for exact details on how Folia
will group nearby chunks.
Each independent region has its own tick loop, which is ticked at the
regular Minecraft tickrate (20TPS). The tick loops are executed
on a thread pool in parallel. There is no main thread anymore, 
as each region effectively has its own "main thread" that executes
the entire tick loop.

For a server with many spread out players, Folia will create many
spread out regions and tick them all in parallel on a configurable sized
threadpool. Thus, Folia should scale well for servers like this.

Folia is also its own project, this will not be merged into Paper
for the foreseeable future. 

A more detailed but abstract overview: [Project overview](https://docs.papermc.io/folia/reference/overview).

**Comment by Q2297045667** [comment]
> Are you talking about Folia the paper fork?

yes
